tests/integration/redis-cli.tcl

```diff
@@ -305,12 +305,29 @@ start_server {tags {"cli"}} {
         for {set i 0} {$i < 100} {incr i} {
            r set test-key test-value-$i
         }
-        r client kill type slave
-        catch {
-            assert_match {*SET*key-a*} [read_cli $fd]
+
+        wait_for_condition 500 100 {
+            [string match {*test-value-99*} [read_cli $fd]]
+        } else {
+            fail "redis-cli --replica didn't read commands"
         }
 
+        r client kill type slave
+        #catch { read_cli $fd } err
+        #assert_match {*exited abnormally*} $err
+
         close_cli $fd
```

Comment:
@oranagra Solved: We put it in blocking mode before it gets killed. Then `close` waits for it to exit and we get the stderr output.

```suggestion
        fconfigure $fd -blocking true
        r client kill type slave
        catch { close_cli $fd } err
        assert_match {*Server closed the connection*} $err
```

From the TCL docs:

* close: "If channelId is a blocking channel for a command pipeline then close waits for the child processes to complete." (https://www.tcl.tk/man/tcl8.5/TclCmd/close.html)
* open: "If the command (or one of the commands) executed in the command pipeline returns an error (according to the definition in exec), a Tcl error is generated when close is called on the channel unless the pipeline is in non-blocking mode then no exit status is returned (a silent close with -blocking 0)." (https://www.tcl.tk/man/tcl8.5/TclCmd/open.html)